src/XML/src/FMI3/fmi3_xml_parser.c

```diff
@@ -1320,6 +1320,13 @@ static void XMLCALL fmi3_parse_element_end(void* c, const char *elm) {
 
     if (context->skipElementCnt) {
         context->skipElementCnt--;
+        // Need to clear buffer
```

Comment:
Wouldn't it make more sense to move this code to the parse_element_start code? In the block where we detect that the handler failed.

Reply:
Good point, it does make more sense. Moved it.